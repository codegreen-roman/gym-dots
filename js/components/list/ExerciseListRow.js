import React from 'react'
import { Exercise } from './Exercise'
import { object } from 'prop-types'
import glamorous from 'glamorous'

const _ExerciseListRow = ({ exercise }) => {
    const Li = glamorous.li({
        padding: 10,
        display: 'flex'
    })
    return (
        <Li>
            <Exercise exercise={exercise} />
        </Li>
    )
}

_ExerciseListRow.propTypes = {
    exercise: object.isRequired
}

export const ExerciseListRow = _ExerciseListRow
