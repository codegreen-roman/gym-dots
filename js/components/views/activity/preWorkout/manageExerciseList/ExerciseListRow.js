import React from 'react'
import { func, string, number } from 'prop-types'
import { exerciseTitle, exerciseDetails } from './ExerciseList.glamor'

const onRowClick = () => {} // noop

const _ExerciseListRow = ({ name, weight, reps, sets, onRowClick, exerciseId }) => {
    return (
        <li data-test={`exercise-row-${exerciseId}`} onClick={onRowClick}>
            <h3 {...exerciseTitle}>
                {name}
            </h3>
            <span {...exerciseDetails}>
                {weight} kg x {reps} reps x {sets} sets
            </span>
        </li>
    )
}

_ExerciseListRow.defaultProps = {
    onRowClick
}

_ExerciseListRow.propTypes = {
    name: string.isRequired,
    weight: number.isRequired,
    reps: number.isRequired,
    sets: number.isRequired,
    onRowClick: func.isRequired
}

export const ExerciseListRow = _ExerciseListRow
