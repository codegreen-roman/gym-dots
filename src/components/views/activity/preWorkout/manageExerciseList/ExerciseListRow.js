import React from 'react'
import { func, string, number } from 'prop-types'
import {
    exerciseRow,
    exerciseDataWrapper,
    exerciseTitle,
    exerciseDetails,
    exerciseElementsWrapper,
    exerciseLine,
    exerciseDot
} from './ExerciseList.glamor'

const onRowClick = () => {} // noop

const _ExerciseListRow = ({
    name,
    weight,
    reps,
    sets,
    exerciseKey,
    onRowClick
}) => {
    return (
        <li {...exerciseRow} data-test={`exercise-row-${exerciseKey}`} onClick={onRowClick}>
            <div {...exerciseElementsWrapper}>
                <div {...exerciseLine} />
                <div {...exerciseDot} />
            </div>
            <div {...exerciseDataWrapper}>
                <div {...exerciseTitle}>
                    {name}
                </div>
                <div {...exerciseDetails} data-test='exercise-details'>
                    {weight} kg x {reps} reps x {sets} sets
                </div>
            </div>
        </li>
    )
}

_ExerciseListRow.defaultProps = {
    onRowClick
}

_ExerciseListRow.propTypes = {
    name: string.isRequired,
    weight: number.isRequired,
    reps: number.isRequired,
    sets: number.isRequired,
    exerciseKey: string.isRequired,
    onRowClick: func.isRequired
}

export const ExerciseListRow = _ExerciseListRow
