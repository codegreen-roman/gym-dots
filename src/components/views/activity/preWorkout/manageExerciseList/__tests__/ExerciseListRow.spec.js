import React from 'react'
import { shallow, mount } from 'enzyme'
import { ExerciseListRow } from '../ExerciseListRow'

const setup = props => {
    const actions = {
        onRowClick: jest.fn()
    }

    const wrapper = shallow(<ExerciseListRow {...props} {...actions} />)
    const mounted = mount(<ExerciseListRow {...props} {...actions} />)

    return {
        wrapper: wrapper,
        mounted: mounted,
        actions: actions,
        exerciseDetails: wrapper.find('[data-test="exercise-details"]')
    }
}

describe('ExerciseListRow component', () => {
    let props
    beforeEach(() => {
        props = {
            name: 'Push-ups',
            exerciseKey: 'zzt',
            sets: 5,
            reps: 20,
            weight: 0,
            onRowClick: jest.fn()
        }
    })

    it('should render exercise', () => {
        const { wrapper } = setup(props)
        expect(wrapper).toMatchSnapshot()
    })

    it('should have matching text', () => {
        const { exerciseDetails } = setup(props)
        expect(exerciseDetails.text()).toMatch(/^0 kg x 20 reps x 5 set/)
    })

    it('should render same props as passed', () => {
        const { mounted } = setup(props)
        expect(mounted.props().sets).toEqual(5)
        expect(mounted.props().weight).toEqual(0)
        expect(mounted.props().reps).toEqual(20)
        expect(mounted.props().name).toEqual('Push-ups')
    })

    describe('Clicking on a ExerciseListRow', () => {
        it('should call actions.onRowClick', () => {
            const { wrapper, actions } = setup(props)
            wrapper.simulate('click')
            expect(actions.onRowClick).toBeCalled()
        })
    })
})
