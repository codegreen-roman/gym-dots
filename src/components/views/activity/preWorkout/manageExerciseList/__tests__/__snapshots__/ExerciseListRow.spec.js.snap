// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExerciseListRow component should render exercise 1`] = `
.css-wd60qt,
[data-css-wd60qt] {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: horizontal;
  -webkit-box-direction: normal;
  -ms-flex-flow: row nowrap;
  flex-flow: row nowrap;
  -webkit-flex-flow: row nowrap;
}

.css-eypxw5,
[data-css-eypxw5] {
  width: 100%;
  margin-left: 30px;
  margin-top: 40px;
}

.css-1vhzv1k,
[data-css-1vhzv1k] {
  font-size: 22px;
  font-weight: 600;
  color: #5E6B7D;
  text-transform: capitalize;
}

.css-1q1fiqx,
[data-css-1q1fiqx] {
  font-size: 14px;
  font-weight: 100;
  color: #919AA7;
}

.css-12buxqa,
[data-css-12buxqa] {
  margin-top: -10px;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
  -ms-flex-flow: column wrap;
  flex-flow: column wrap;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: #fff;
  -webkit-flex-flow: column wrap;
  -webkit-align-items: center;
}

.css-ikdvzh,
[data-css-ikdvzh] {
  height: 55px;
  width: 3px;
  background: #83D1F3;
  border-radius: 3px;
}

.css-1hvluh9,
[data-css-1hvluh9] {
  border-radius: 50%;
  width: 8px;
  height: 8px;
  border: 3px solid #46B4E3;
  margin: 4px 0;
}

<li
  data-css-wd60qt=""
  data-test="exercise-row-zzt"
  onClick={[Function]}
>
  <div
    data-css-12buxqa=""
  >
    <div
      data-css-ikdvzh=""
    />
    <div
      data-css-1hvluh9=""
    />
  </div>
  <div
    data-css-eypxw5=""
  >
    <div
      data-css-1vhzv1k=""
    >
      Push-ups
    </div>
    <div
      data-css-1q1fiqx=""
      data-test="exercise-details"
    >
      0
       kg x 
      20
       reps x 
      5
       sets
    </div>
  </div>
</li>
`;
